### ISSUE TITLE
docs say vue, axios, and lodash are peer dependencies, but they are not

### ISSUE THREAD

**Issue by djsmedes** [FiguredLimited/vue-mc#109]
in package.json, we have
```
"dependencies": {
        "axios": "^0.18.0",
        "date-fns": "^2.0.0-alpha.25",
        "lodash": "^4.17.11",
        "validator": "^10.9.0",
        "vue": "2.5.17"
},
```

Should these not be `peerDependencies` ? There is something going on with npm and/or yarn that I don't quite understand. I have a project using vue-mc, and in that project's `node_modules/vue-mc/` folder, there is another `node_modules` folder that contains `vue`. This strangely enough makes `vue.runtime.esm` get included in my project's built files twice, once from the project itself and once from vue-mc. However if I just delete `node_modules/vue-mc/node_modules` things work as I'd expect.

What my build looks like in current state - note the doubling-up on `vue.runtime.esm`:
![current_state](https://user-images.githubusercontent.com/7870631/53674569-d46bc080-3c5c-11e9-9539-56e3daac6e6a.png)
This is 330.30 KiB parsed size.

What happens if I delete vue from vue-mc's internal `node_modules` folder:
![no_double_vue](https://user-images.githubusercontent.com/7870631/53674657-6ecc0400-3c5d-11e9-92df-a2b4f2e2580c.png)
269.54 KiB parsed size.

_Excerpted from original post by @djsmedes in https://github.com/FiguredLimited/vue-mc/issues/89#issuecomment-468865217_

**Comment by AndreasHerss** [comment]
i got this issue too, funny enough axios and lodash doesn't double up.

Here, vue is double, at a total bundle size of 340kb parsed
![2019-03-28 10_32_29-Webpack Bundle Analyzer – Google Chrome](https://user-images.githubusercontent.com/6894019/55146641-45f83b00-5145-11e9-9e92-f17433fe09b8.png)

and here i tried your trick removing the node_modules from vue-mc, vue shows up only once, and a parsed size of 255kb.
![2019-03-28 10_34_28-](https://user-images.githubusercontent.com/6894019/55146752-74761600-5145-11e9-9692-5a8a2e5b345c.png)


but thanks for the trick about removing node_modules ;)

**Comment by djsmedes** [comment]
You’re welcome! Unfortunately I doubt it will work in production environments if you’re running on a hosted service where you don’t have access to just go in and delete a folder (I use Heroku, for instance).

Switching vue to a peerDependency should resolve that issue. Note you still have to have vue installed locally alongside vue-mc or the tests won’t run, but yarn doesn’t have a way to have a package installed without including it in package.json (or, as far as I can tell, proper support for peerDependencies at all). So in order to make this work, I’ve manually edited package.json to make vue (and lodash and axios) peerDependencies and then run `npm install —no-save vue axios lodash`

**Comment by rtheunissen** [comment]
Would it help if we made them dev dependencies only? I wonder if you could set both peer and dev dependencies.. otherwise would proper version constraints solve this without the need for peer dependencies?

**Comment by djsmedes** [comment]
It's been a while since I've directly messed with this. I will try to do some experiments this weekend. Issues with package size aside, the biggest problem that can arise here is if vue-mc is using its own instance of axios, because in my other project which is making use of vue-mc, I have set up some defaults on the axios instance for authentication; if vue-mc uses its local axios, all the API calls fail with 403 errors.

I remember at the time I was working on this a few months ago, there was a recommendation I found somewhere to set them as peerDependencies _and_ devDependencies, but that I discarded this suggestion for some reason :man_shrugging: 

Like I said, I will experiment a little this weekend.

**Comment by rtheunissen** [comment]
I have removed the documentation about peer dependencies for now as part of [v0.6.0](https://github.com/FiguredLimited/vue-mc/releases/tag/v0.6.0).

**Comment by djsmedes** [comment]
Apologies for not getting to this. I think there might still be problems around this area - as an example, I'm doing this in my main app -

```
import axios from "axios";
// ...
axios.defaults.xsrfCookieName = "csrftoken";
axios.defaults.xsrfHeaderName = "X-CSRFToken";
```

But if I'm running from `vue-mc` locally and it has its own axios install, then `Model.save()` & co. will use _that_ axios instance, which doesn't have those csrf settings, and my backend will reject it.

Anyway I will open a new issue when I have time to experiment a bit. Obviously running from a local install is not the actual success criterion for the package. Thanks for looking in on this!

**Comment by albertpratomo** [comment]
> But if I'm running from `vue-mc` locally and it has its own axios install, then `Model.save()` & co. will use _that_ axios instance, which doesn't have those csrf settings, and my backend will reject it.

Just ran into the exact situation here ☝️ . Totally agree that at least Vue and Axios should be made peer dependencies.
